contracts/zero-ex/contracts/src/features/UniswapV2Feature.sol

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+/*
+
+  Copyright 2021 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.6.5;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/v06/IERC20TokenV06.sol";
+import "@0x/contracts-erc20/contracts/src/v06/IEtherTokenV06.sol";
+import "../migrations/LibMigrate.sol";
+import "../fixins/FixinCommon.sol";
+import "./interfaces/IFeature.sol";
+import "./interfaces/IPancakeSwapFeature.sol";
+import "@uniswap/v2-periphery/contracts/libraries/UniswapV2Library.sol";
+
+/// @dev VIP UniswapV2 fill functions.
+contract UniswapV2Feature is
+    IFeature,
+    IPancakeSwapFeature,
+    FixinCommon
+{
+    /// @dev Name of this feature.
+    string public constant override FEATURE_NAME = "UniswapV2Feature";
+    /// @dev Version of this feature.
+    uint256 public immutable override FEATURE_VERSION = _encodeVersion(1, 0, 0);
+
+    // TODO: complete the list
+    // address of the SushiSwapV2 factory contract.
+    address constant private SUSHISWAPV2_FACTORY = address(0xc35DADB65012eC5796536bD9864eD8773aBc74C4);
+
+    // other available
+
+    function _getFactory(ProtocolFork fork) private view returns(address) {
+        // TODO: complete the list
+        if (fork == ProtocolFork.SushiSwap) {
+            return SUSHISWAPV2_FACTORY;
+        } else {
+            revert("isnt it weired revert in view fun?");
+        }
+    }
+
+    function sellToUniswapV2(
+        IERC20TokenV06[] calldata tokens,
+        uint256 sellAmount,
+        uint256 minBuyAmount,
+        ProtocolFork fork
+    )
+        external
+        payable
```

Comment:
Payable implies that this function can also sell ETH, right? But I don't see that case being handled.

Reply:
Following the assembly version of this feature:
- if we detect `0xeee...` for the first token, it's an ETH sell and we need to wrap ETH->WETH and transfer from ourselves.
- If we detect `0xeee...` for the last token, it's an ETH buy and we need to make ourselves the final recipient in the pair swap, unwrap the bought WETH->ETH, then transfer that to `msg.sender`.